src/lib/cmdline.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of clr-boot-manager.
+ *
+ * Copyright © 2017 Intel Corporation
+ *
+ * clr-boot-manager is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ */
+
+#define _GNU_SOURCE
+
+#include "cmdline.h"
+#include "config.h"
+#include "log.h"
+#include "nica/files.h"
+#include "util.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <glob.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+static bool cbm_parse_cmdline_file_internal(const char *path, FILE *out)
+{
+        autofree(FILE) *f = NULL;
+        size_t sn;
+        ssize_t r = 0;
+        char *buf = NULL;
+        bool ret = true;
+        bool hadcontent = false;
+
+        f = fopen(path, "r");
+        if (!f) {
+                if (errno != ENOENT) {
+                        LOG_ERROR("Unable to open %s: %s", path, strerror(errno));
+                }
+                return NULL;
+        }
+
+        while ((r = getline(&buf, &sn, f)) > 0) {
+                ssize_t cur = 0;
+                autofree(char) *value = NULL;
+
+                /* Strip newlines */
+                if (r >= 1 && buf[r - 1] == '\n') {
+                        buf[r - 1] = '\0';
+                        --r;
+                }
+                char *l = buf;
+
+                /* Skip empty lines */
+                if ((r == 1 && buf[0] == '\n') || r < 1) {
+                        goto next_line;
+                }
+
+                /* Skip the starting whitespace */
+                while (isspace(*l)) {
+                        ++l;
+                        if (!*l) {
+                                break;
+                        }
+                }
+
+                /* Skip a comment */
+                if (l[0] == '#') {
+                        goto next_line;
+                }
+
+                cur = (l - buf);
+                r -= cur;
+
+                /* For successive new lines, add a space before writing anything. */
+                if (hadcontent) {
+                        if (fwrite(" ", 1, 1, out) != 1) {
+                                ret = false;
+                                break;
+                        }
+                        hadcontent = false;
```

Comment:
So I maybe don't understand hadcontent super well. I think the idea should be "if this is the first thing being written don't add the space, otherwise add the space". Based on that setting hadcontent false in the body doesn't make sense (and after the write below is set to true unconditionally).

I do think that there should be a continue before this block to see if r == 0 and goto next_line if so which maybe resolves something that hadcontent was supposed to take care of?